### ISSUE TITLE
activeDeadlineSeconds like capability in a job

### ISSUE THREAD

**Issue by AlexFly666** [volcano-sh/volcano#2761]
#### What would you like to be added:

A activeDeadlineSeconds-like capability in a job that limits the job survival time
The job is not finished within the specified time

#### Why is this needed:

Avoid some long-running jobs that do not set up the completed exit mechanism

**Comment by wangyang0616** [comment]
There is a definition of ActiveDeadlineSeconds in the job of k8s, which can control the running time of the job. If Volcano also adds corresponding definitions and functions, users will be more flexible in using it. I vote for it.

**Comment by hwdef** [comment]
still need this

**Comment by fengruotj** [comment]
I am very interested in this issue, please assign it to me.  /assigin

**Comment by hwdef** [comment]
Please submit a design document first.
We also had some discussions during last week's weekly meeting:
1. Add more scenarios. (You can refer to kubernetes job)
2. After the set time is reached, the vcjob status changes to complete

**Comment by fengruotj** [comment]
> Please submit a design document first. We also had some discussions during last week's weekly meeting:
> 
> 1. Add more scenarios. (You can refer to kubernetes job)
> 2. After the set time is reached, the vcjob status changes to complete


ok, I got it, where is week's weekly meeting. I am very interested to join you.

**Comment by hwdef** [comment]
https://zoom.us/j/91804791393
Every Friday at 3pm

You can also add WeChat account k8s2222 (华为云小助手2号) to let her pull you into the group.
Meeting announcements will be posted in the group

**Comment by hwdef** [comment]
@fengruotj 
Did you still work on this?

**Comment by hwdef** [comment]
@fengruotj 
Yes, you can. welcome

**Comment by elysium-w** [comment]
@hwdef I still don't know much about volcano source code, can you please help me clarify which functions this change will involve?or anything I should deep learn about volcano

**Comment by hwdef** [comment]
two parts:
1. Modify api, I already submit a PR https://github.com/volcano-sh/apis/pull/138
2. Modify controller code, The logic is similar to the job native to k8s, and we can learn from the logic of k8s

**Comment by hwdef** [comment]
First we need to submit a design document

**Comment by elysium-w** [comment]
@hwdef sure,I need to comb the source code, determine the specific scope of impact

**Comment by Bhukya-Sriram** [comment]
Hi, I’m a new contributor and would like to work on this issue. Can I take it?

**Comment by hwdef** [comment]
@elysium-w 
Are you still doing this?

**Comment by elysium-w** [comment]
> [@elysium-w](https://github.com/elysium-w) Are you still doing this?

yes,I was busy in those month for find job and graduation,right now I already finish it temporarily,sorry for my slow progress,I'll try to finish it these days,no more than 5 days. I'm so sorry.

**Comment by meekailkhan** [comment]
looks like no one is here is solved it yet. If it still available i'd be happy to work on it
/assign
